src/rust/receiver-mock/src/metadata.rs

```diff
@@ -1,8 +1,41 @@
+use actix_web::http::header::HeaderMap;
+use actix_web::http::HeaderValue;
 use anyhow::anyhow;
 use std::collections::HashMap;
 
 pub type Metadata = HashMap<String, String>;
 
+pub fn get_metadata_from_headers(headers: &HeaderMap) -> Result<Metadata, anyhow::Error> {
+    let empty_header = HeaderValue::from_str("").unwrap();
+
+    let x_sumo_fields_value = headers.get("x-sumo-fields").unwrap_or(&empty_header).to_str().unwrap();
```

Comment:
Maybe just return an empty result if `headers.get("x-sumo-fields")` returns none?

Reply:
I need to create the metadata map in case any of the other fields are set anyway, and the parser handles an empty string the right way, so I think this is the simplest way.

Reply:
> and the parser handles an empty string the right way, so I think this is the simplest way

It might be but to a reader (like myself) knowing that `.get()` returns an option, that's a perfect place to check if we actually have the header. If not then don't bother even thinking about parsing it or even worse relying on an implicit contract that "the parser will do the right thing".

Reply:
If you think it'll make the code easier to understand, I'll change it. Note that there's no implicit contract here, the parser has to be able to deal with the header containing an empty string correctly. The implicit assumption is, at most, that the lack of a header and an empty value have the same meaning.

Reply:
> The implicit assumption is, at most, that the lack of a header and an empty value have the same meaning.

Exactly. Net effect is the same but IMHO those should happen at the very least in separate branches.